File: ruby/Grace.rb
#!/usr/bin/env ruby
#Commentaire

FILENAME = "Grace_kid.rb"

QUINE = "#!/usr/bin/env ruby%{nl}#Commentaire%{nl}%{nl}FILENAME = %{bq}Grace_kid.rb%{bq}%{nl}%{nl}QUINE = %{bq}%{quine}%{bq}%{nl}%{nl}File.open(FILENAME, 'w') do |f|%{nl}%{ht}f.puts QUINE %% {nl: 10.chr, ht: 9.chr, bq: 34.chr, quine: QUINE}%{nl}end"

File.open(FILENAME, 'w') do |f|
  f.puts QUINE % {nl: 10.chr, ht: 9.chr,bq: 34.chr, quine: QUINE}
end

File: python/Grace.py
#!/usr/bin/env python3
#Commentaire

FILENAME = "Grace_kid.py"

QUINE = "#!/usr/bin/env python3{0}#Commentaire{0}{0}FILENAME = {2}Grace_kid.py{2}{0}{0}QUINE = {2}{3}{2}{0}{0}{0}with open(FILENAME, 'w') as f:{0}{1}f.write(QUINE.format(chr(10), chr(9), chr(34), QUINE))"

with open(FILENAME, 'w') as f:
    f.write(QUINE.format(chr(10), chr(9), chr(34), QUINE))
